trio/_core/tests/tutil.py

```diff
@@ -97,7 +98,11 @@ def restore_unraisablehook():
 
     @contextmanager
     def disable_threading_excepthook():
-        threading.excepthook, prev = _noop, threading.excepthook
+        if sys.version_info >= (3, 10):
+            threading.excepthook, prev = threading.__excepthook__, threading.excepthook
+        else:
+            threading.excepthook, prev = _noop, threading.excepthook
+
```

Comment:
This is mysterious :-). What's going on here?

Reply:
I think this was for disabling the thread hooks that pytest started using in 6.2 or so to catch and complain about uncaught exceptions in threads.  In certain cases, IIRC, we literally have explanations in the code, or even the error messages, saying that yes, these will fall out un-handled in the thread or whatever but that that's the way it is and we know.  But @graingert did this bit.  The added bit here is to use the 'original' excepthook instead of a noop when it is available (in 3.10).  This additional feature in 3.10 let's this code be more consistent with the `restore_unraisablehook()` above.

Hopefully that's in the direction you were wondering about...